app/page/home/activity.tsx

```diff
@@ -51,181 +64,149 @@ const UserOpHistoryItem: React.FC<{
-            to: 0x0000...00000
-          </div>
+      <div className="w-full flex items-center py-[13px] justify-between">
+        <div className="flex flex-col items-start gap-[8px]">
+          {/* Activity Topic element */}
+          {topic[topicType]}
+          {/* Activity Time element */}
+          <Time date={creationDate} time={creationTime} link={link} />
         </div>
-        {/* Transaction time */}
-        <div>
-          <div className="flex flex-col items-end">
-            <div className="mb-[4px] leading-[19px] text-[16px] text-[#000000]">
-              {creationDate}
-            </div>
-            <div className="leading-[19px] text-[16px] text-[#000000]">
-              {creationTime}
-            </div>
-          </div>
-        </div>
-      </a>
+        {/* Activity Status element */}
+        <Status status={status} tokenInfo={tokenInfo} />
+      </div>
     )
   }
 
-  if (txLog.status === TransactionStatus.Rejected) {
-    return (
-      <div className="w-full flex items-center p-[14px_0px_14px_0px]">
-        <div className="flex-grow flex flex-col items-start">
-          {/* Status and image */}
+  // Others' case
+  return (
+    <div className="w-full flex items-center py-[13px] justify-between">
+      <div className="flex flex-col items-start gap-[8px]">
+        {/* Activity Topic element */}
+        {topic[topicType]}
+        {/* Activity Time element */}
+        <Time date={creationDate} time={creationTime} />
+      </div>
+      {/* Activity Status element */}
+      <Status status={status} />
+    </div>
+  )
+}
+
+type TopicType = "send" | "contract" | "receive"
+
+const TopicTemplate: React.FC<{
+  Icon: React.ComponentType<{ className?: string }>
+  topic: string
+}> = ({ Icon, topic }) => {
+  return (
+    <div className="flex items-center">
+      <Icon className="w-[16px] h-[16px] mr-[8px]" />
+      <div className="text-[16px] text-[#000000]">{topic}</div>
+    </div>
+  )
+}
+
+const topic: Record<TopicType, JSX.Element> = {
+  send: <TopicTemplate Icon={ArrowUpRight} topic="Send" />,
+  contract: (
+    <TopicTemplate Icon={ArrowRightArrowLeft} topic="Contract interaction" />
+  ),
+  receive: <TopicTemplate Icon={ArrowDownLeft} topic="Receive" />
+}
+
+const Status: React.FC<{
+  status: TransactionStatus
+  tokenInfo?: TokenTransferInfo
+}> = ({ status, tokenInfo }) => {
+  if (tokenInfo) {
+    switch (status) {
+      case TransactionStatus.Succeeded:
+        return (
           <div className="flex items-center">
-            <OctagonXmark className="w-[16px] h-[16px] mr-[8px]" />
-            <div className="mr-[8px] leading-[19px] text-[16px] text-[#000000]">
-              {status}
+            <div className="text-[16px] text-[#FF5151] whitespace-nowrap">
+              - {number.formatUnitsToFixed(tokenInfo.value, 18, 4)}{" "}
+              {tokenInfo.symbol}
             </div>
           </div>
+        )
+      // TODO: Handle receive case
```

Comment:
Does **receive case** have `TransactionStatus`?

Reply:
No, `TransactionStatus` is not part of the `Account` life cycle.
It should be another status for listening to incoming transactions on new blocks.